assets/js/blocks/product-categories/index.js

```diff
@@ -105,6 +121,7 @@ registerBlockType( 'woocommerce/product-categories', {
 			save( props ) {
 				const {
 					hasCount,
+					hasImage,
```

Comment:
you shouldn't need any of the new `hasImage` attributes in the `deprecated` block as it's never been in a released block.